azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,360 @@
+            else:
+                # File is not present in the archive, add it
+                LogCollector._log_to_results_file("Updating archive, adding new file {0}".format(archive_file_name))
+                LogCollector._add_file_to_archive(file_name, archive_file_name)
+
+    def _parse_manifest_file(self):
+        files_to_collect = set()
+        manifest_entries = self._read_manifest_file()
+
+        for entry in manifest_entries:
+            # The entry can be one of the four flavours:
+            # 1) ll,/etc/udev/rules.d -- list out contents of the folder and store to results file
+            # 2) echo,### Gathering Configuration Files ### -- print message to results file
+            # 3) copy,/var/lib/waagent/provisioned -- add file to list of files to be collected
+            # 4) diskinfo, -- ignore other commands for now
+
+            contents = entry.split(",")
+            if len(contents) != 2:
+                continue
+
+            command, value = entry.split(",")
+
+            if command == "ll":
+                self._parse_ll_command(value)
+            elif command == "echo":
+                self._parse_echo_command(value)
+            elif command == "copy":
+                files_to_collect.update(self._parse_copy_command(value))
+
+        return files_to_collect
+
+    @staticmethod
+    def _truncate_large_file(file_path):
+        # Truncate large file to size limit (keep freshest entries of the file), copy file to a temporary location
+        # and update file path in list of files to collect
+        try:
+            # Binary files cannot be truncated, don't include large binary files
+            if os.path.splitext(file_path)[1] == ".gz":
+                LogCollector._log_to_results_file("Discarding large binary file {0}".format(file_path))
+                return None
+
+            truncated_file_path = os.path.join(TRUNCATED_FILES_DIR, file_path.replace(os.path.sep, "_"))
+            if os.path.exists(truncated_file_path):
+                original_file_mtime = os.path.getmtime(file_path)
+                truncated_file_mtime = os.path.getmtime(truncated_file_path)
+
+                # If the original file hasn't been updated since the truncated file, it means there were no changes
+                # and we don't need to truncate it again.
+                if original_file_mtime < truncated_file_mtime:
+                    return truncated_file_path
+
+            # Get the last N bytes of the file
+            with open(truncated_file_path, "w+") as fh:
+                command_string = "tail -c {0} {1}".format(FILE_SIZE_LIMIT, file_path)
+                command = safe_shlex_split(command_string)
+                run_command(command, stdout=fh)
+
+            return truncated_file_path
+        except OSError as e:
+            LogCollector._log_to_results_file("Failed to truncate large file: {0}".format(ustr(e)))
```

Comment:
This module shouldn't depend on the agent's `logger` or waagent.log, it is a standalone module. The results file is added to the zip (even in case of unsuccessful log collection).